guides/source/working_with_javascript_in_rails.md

```diff
@@ -300,3 +300,38 @@ added to the form that the `button_to` helper renders internally:
 ```erb
 <%= button_to "Delete post", post, method: :delete, form: { data: { turbo_confirm: "Are you sure?" } } %>
 ```
+
+### Ajax Requests
+
+When making non-GET requests from JavaScript the `X-CSRF-Token` header is required.
+Without this header requests won't be accepted by Rails.
+
+[Rails Request.JS](https://github.com/rails/request.js) encapsulates the logic
+of adding the request headers that are required by Rails.
+
+Just import the FetchRequest class from the package and instantiate it passing
+the request method, url, options, then call `await request.perform()` and do what
+do you need with the response.
+
+Example:
+
+```javascript
+import { FetchRequest } from '@rails/request.js'
+
+....
+
+async myMethod () {
+  const request = new FetchRequest('post', 'localhost:3000/my_endpoint', {
+    body: JSON.stringify({ name: 'Request.JS' })
+  })
+  const response = await request.perform()
+  if (response.ok) {
+    const body = await response.text
+  }
+}
+```
+
+When using another library to make Ajax calls, it is necessary to add the
+security token as a default header for Ajax calls in your library. To get the
+token, have a look at <meta name='csrf-token' content='THE-TOKEN'> tag printed
+by <%= csrf_meta_tags %> in your application view.
```

Comment:
I don't think users should be copy/pasting the output of `csrf_meta_tags` in their code as it will break the next time the secret changes.

I'm not sure the right approach is here though off the top of my head.

Reply:
`document.head.querySelector("meta[name=csrf-token]")?.content` does the trick ([adapted from here](https://github.com/rails/request.js/blob/85d16d3ee0990e67651a025e32abdecfc42e9bb6/src/lib/utils.js#L29))

Reply:
Thanks @bdewater !
I copied the original text from the security guide. I'll change it there as well.